tests/garbage_collector_test.py

```diff
@@ -0,0 +1,66 @@
+"""Tests for aws/garbage_collector."""
+
+import os
+import unittest
+from mock import MagicMock as Mock
+from treadmill_aws import garbage_collector
+
+
+# pylint: disable=protected-access
+class GarbageCollector(unittest.TestCase):
+    """Tests garbage_collector"""
+
+    def setUp(self):
+        """Set up mocks"""
+        self.ec2_instances = {
+            'Reservations': [
+                {'Instances': [
+                    {'Tags': [{"Key": "Name",
+                               "Value": 'valid-instance.foo.bar.com'}]}]}]}
+        self.valid_instances = {"valid-instance.foo.bar.com"}
+        self.ldap_servers = {"valid-instance.foo.bar.com",
+                             "terminated-instance.foo.bar.com"}
+        self.ipa_servers = self.ldap_servers
+
+        self.admin_srv = Mock()
+        self.admin_srv.admin = Mock()
+        self.admin_srv.admin.search = Mock()
+        self.admin_srv.delete = Mock()
+
+    def tearDown(self):
+        """Tear down"""
+        os.environ["IPA_DOMAIN"] = ""
```

Comment:
do not mess with environment in tests, it can have side effects (and in the code you do not need them anyways, so it is easy fix).